packages/headless/src/controllers/facets/category-facet/headless-category-facet.ts

```diff
@@ -35,22 +37,199 @@ import {validateOptions} from '../../../utils/validate-payload';
 import {
   CategoryFacetOptions,
   categoryFacetOptionsSchema,
+  CategoryFacetSearchOptions,
 } from './headless-category-facet-options';
 import {determineFacetId} from '../_common/facet-id-determinor';
+import {FacetValueState} from '../../../features/facets/facet-api/value';
+
+export {CategoryFacetOptions, CategoryFacetSearchOptions};
 
-export {CategoryFacetOptions};
-export type CategoryFacetProps = {
+export interface CategoryFacetProps {
   /** The options for the `CategoryFacet` controller. */
   options: CategoryFacetOptions;
-};
+}
+
 /**
  * The `CategoryFacet` headless controller offers a high-level interface for designing a facet UI controller that renders values in a hierarchical fashion.
  */
-export type CategoryFacet = ReturnType<typeof buildCategoryFacet>;
+export interface CategoryFacet extends Controller {
+  /**
+   * Provides methods to search the facet's values.
+   */
+  facetSearch: CategoryFacetSearch;
+
+  /**
+   * Toggles the specified facet value.
+   *
+   * @param selection The facet value to toggle.
+   */
+  toggleSelect(selection: CategoryFacetValue): void;
+
+  /**
+   * Deselects all facet values.
+   * */
+  deselectAll(): void;
+
+  /**
+   * Sorts the facet values according to the specified criterion.
+   *
+   * @param criterion - The criterion to sort values by.
+   */
+  sortBy(criterion: CategoryFacetSortCriterion): void;
+
+  /**
+   * Checks whether the facet values are sorted according to the specified criterion.
+   *
+   * @param criterion - The criterion to compare.
+   * @returns Whether the facet values are sorted according to the specified criterion.
+   */
+  isSortedBy(criterion: CategoryFacetSortCriterion): boolean;
+
+  /**
+   * Increases the number of values displayed in the facet to the next multiple of the originally configured value.
+   */
+  showMoreValues(): void;
+
+  /**
+   * Sets the displayed number of values to the originally configured value.
```

Comment:
`Sets the displayed number of values` -> `Sets the number of values displayed in the facet`